tests/unit/hook_packages/terraform/hooks/prepare/test_resource_linking.py

```diff
@@ -131,3 +132,144 @@ def test_resource_full_address_root_module(self):
         module = TFModule(None, None, {}, {}, {}, {})
         resource = TFResource("resource_address", "type", module, {})
         self.assertEqual(resource.full_address, "resource_address")
+
+    def test_resolve_module_variable_constant_value(self):
+        constant_value = ConstantValue(value="layer.arn")
+        module = TFModule(
+            variables={"layer": constant_value},
+            resources=[],
+            child_modules={},
+            outputs={},
+            full_address="",
+            parent_module=None,
+        )
+        results = _resolve_module_variable(module, "layer")
+        expected_result = "layer.arn"
+        for result in results:
+            self.assertEqual(result.value, expected_result)
```

Comment:
I am wondering why do you do it in loop!! .. we should expect only one item in this list. I think we should add an assert to validate that the list length is only 1 and it contains the value we expect.